lms/djangoapps/teams/templates/teams/teams.html

```diff
@@ -44,8 +44,8 @@
         teamMembershipDetailUrl: '${ team_membership_detail_url }',
         myTeamsUrl: '${ my_teams_url }',
         maxTeamSize: ${ course.teams_max_size },
```

Comment:
I like proposal 2, because it means you can treat the number as a number, rather than having explicit quotes in the template turning it into a string, and explicit code in the client turning it back into a number.

I don't mind that "escape_js_number" isn't escaping.  It might mean that "escape_" as a prefix is not great because it leaks the implementation.  The point of this function is to make sure the thing is right as a JavaScript number, so maybe "js_number" is enough.